internal/services/hybridcompute/arc_machine_extension_resource.go

```diff
@@ -0,0 +1,549 @@
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("automatic_upgrade_enabled") {
+				properties.Properties.EnableAutomaticUpgrade = &model.EnableAutomaticUpgrade
+			}
+
+			if metadata.ResourceData.HasChange("force_update_tag") {
+				if model.ForceUpdateTag != "" {
+					properties.Properties.ForceUpdateTag = &model.ForceUpdateTag
+				} else {
+					properties.Properties.ForceUpdateTag = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("protected_settings") {
+				var protectedSettingsValue interface{}
+				err := json.Unmarshal([]byte(model.ProtectedSettings), &protectedSettingsValue)
+				if err != nil {
+					return err
+				}
+
+				properties.Properties.ProtectedSettings = &protectedSettingsValue
+			}
+
+			if metadata.ResourceData.HasChange("publisher") {
+				if model.Publisher != "" {
+					properties.Properties.Publisher = &model.Publisher
+				} else {
+					properties.Properties.Publisher = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("settings") {
+				var settingsValue interface{}
+				err := json.Unmarshal([]byte(model.Settings), &settingsValue)
+				if err != nil {
+					return err
+				}
+
+				properties.Properties.Settings = &settingsValue
+			}
+
+			if metadata.ResourceData.HasChange("type") {
+				if model.Type != "" {
+					properties.Properties.Type = &model.Type
+				} else {
+					properties.Properties.Type = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("type_handler_version") {
+				if model.TypeHandlerVersion != "" {
+					properties.Properties.TypeHandlerVersion = &model.TypeHandlerVersion
+				} else {
+					properties.Properties.TypeHandlerVersion = nil
+				}
+			}
+
+			properties.SystemData = nil
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = &model.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ArcMachineExtensionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.HybridCompute.MachineExtensionsClient
+
+			id, err := machineextensions.ParseExtensionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
```

Comment:
We don't need to return an error if Model is nil (same applies to Properties) and for readability we assign it to a variable

```suggestion
			if model := resp.Model; model != nil {
				...
				if props := model.Properties; props != nil {
				...
			
```